src/wamr/wamr.cc

```diff
@@ -118,18 +121,28 @@ class Wamr : public WasmVm {
   std::unordered_map<std::string, WasmFuncPtr> module_functions_;
 };
 
-bool Wamr::load(std::string_view bytecode, std::string_view /*precompiled*/,
+bool Wamr::load(std::string_view bytecode, std::string_view precompiled,
                 const std::unordered_map<uint32_t, std::string> & /*function_names*/) {
   store_ = wasm_store_new(engine());
   if (store_ == nullptr) {
     return false;
   }
 
-  wasm_byte_vec_t binary = {.size = bytecode.size(),
-                            .data = (char *)bytecode.data(),
-                            .num_elems = bytecode.size(),
-                            .size_of_elem = sizeof(byte_t),
-                            .lock = nullptr};
+  wasm_byte_vec_t binary = {0};
+  if (precompiled.empty()) {
+    binary.size = bytecode.size();
+    binary.data = (char *)bytecode.data();
+    binary.num_elems = bytecode.size();
+    binary.size_of_elem = sizeof(byte_t);
+    binary.lock = nullptr;
+  } else {
+    binary.size = precompiled.size();
+    binary.data = (char *)precompiled.data();
+    binary.num_elems = precompiled.size();
+    binary.size_of_elem = sizeof(byte_t);
+    binary.lock = nullptr;
+  }
```

Comment:
IMHO, it would be better to remove AOT support out of this PR and instead add it in #380, since it seems like this doesn't work because of padding anyway.